### PR TITLE
Starting an updated test suite

### PR DESCRIPTION
**Pull request by gknedo**
Hi, 
Based by the comments on an very old [krln](http://github.com/krln) pull request #56 i made an updated one.

What was done and what next:

- [x] Added rspec for tests
- [x] Added Combustion so we dont need a full rails dummy application
- [x] Docker for easily setup
- [x] Factory Bot and an simple test/factory for reference
- [x] FFaker for Semantic Tests and to estimulate good practices
- [ ] More basic tests, so we can have more courage to modify or refactor the gem
- [ ] Setups and generators tests
- [ ] A more complex suite configuration to run tests over diferent setups sceneries

Anyway, we already have something to starting a good test suit :smiley: .

### CODE REVIEW

**Comment by gknedo** [comment]
@ankane ?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,6 +26,26 @@ If the above steps don’t help, create an issue. Include:
 - Detailed steps to reproduce
 - Complete backtraces for exceptions
 
+## Setup
```
Comment (ankane) [inline CONTRIBUTING.md]: Don't think we should use Docker.

File: spec/internal/config/database.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+test:
+  adapter: mysql2
```
Comment (ankane) [inline spec/internal/config/database.yml]: Let's use SQLite instead of MySQL.

File: spec/internal/config/initializers/safely.rb [hunk 3]
```diff
@@ -0,0 +1 @@
+Safely.report_exception_method = -> (e) { raise e }
```
Comment (ankane) [inline spec/internal/config/initializers/safely.rb]: Safely raises errors in test environment by default, so this isn't needed.

File: ahoy_matey.gemspec [hunk 4]
```diff
@@ -28,9 +28,11 @@ Gem::Specification.new do |spec|
 
   spec.add_development_dependency "bundler"
   spec.add_development_dependency "rake"
-  spec.add_development_dependency "minitest"
   spec.add_development_dependency "activerecord"
-  spec.add_development_dependency "pg"
```
Comment (ankane) [inline ahoy_matey.gemspec]: The existing dependencies are still needed for existing test suite.

File: Gemfile [hunk 5]
```diff
@@ -2,5 +2,3 @@ source "https://rubygems.org"
 
 # Specify your gem's dependencies in ahoy.gemspec
 gemspec
-
-gem "rails", "~> 5.1.0"
```
Comment (ankane) [inline Gemfile]: Not sure I understand why this is needed.

File: ahoy_matey.gemspec [hunk 6]
```diff
@@ -28,9 +28,11 @@ Gem::Specification.new do |spec|
...
-  spec.add_development_dependency "mysql2"
-  spec.add_development_dependency "mongoid"
+  spec.add_development_dependency "activejob"
+  spec.add_development_dependency 'mysql2', "~> 0.4.4"
+  spec.add_development_dependency 'combustion'
+  spec.add_development_dependency 'rspec-rails'
+  spec.add_development_dependency 'factory_bot_rails'
```
Comment (ankane) [inline ahoy_matey.gemspec]: Prefer to keep dependencies to minimum, unless they add a lot of value.

File: spec/internal/db/.gitignore [hunk 7]
```diff
@@ -0,0 +1 @@
+*.sqlite
```
Comment (ankane) [inline spec/internal/db/.gitignore]: Let's use existing .gitignore

**Comment by ankane** [comment]
Closing as stale. Feel free to reopen/create a new PR if you address the comments above.

**Pull request closed without merging**